### ISSUE TITLE
check_spe() considerations

### ISSUE THREAD

**Issue by jvelezmagic** [LieberInstitute/spatialLIBD#11]
Hi Leo.
Meeting Comments:

1. `check_spe()` should consider the requested `assaynames` and not default ones.
2. Probably `barcode` is now in `colData` and not in `spatialData`.
3. Describe the request to have the `key` column as a requirement.
4. Describe the special columns `gene_id`,` gene_name`, and `gene_search`.
5. Describe how to define the `Layer` column (e.g. filled of "NA").

**Comment by jvelezmagic** [comment]
Suggestions from today's meeting:
- Include generalization of labels (e.g., discrete variables, continuous variables).
- Generalization of special variables (e.g., SPATIALLIBD, logcounts).
- Notes for the user about the min_count parameter.
- Personal note: Reduced dimensions must have only the first two dimensions.

**Comment by lcolladotor** [comment]
* We no longer default to `counts` and `logcounts` and instead use the `assayNames(spe)`.
* `Barcode` has been fixed.
* We describe `key`, `gene_id`, `gene_name`, `gene_search` in the new vignette.
* We changed `Layer` by `ManualAnnotation` and described in the vignette. 
* We added a few help texts and changed the names of some of the main toggles. We also updated the `documentation_spe.md` file.
* We included a little help text box about the min count.
* At `documentation_spe.md` we mention that only the first 2 dimensions will be shown. The internal code can handle more dimensions now. Note that if no reduced dims are provided, the app will still work now.
